### DISCUSSION TITLE
v3.1.0 - property conductor.additional.modules not found

### DISCUSSION THREAD

**Discussion by sukruthiarjula** [Netflix/conductor#2397] [Q&A]
Hi,

We are on 2.31.4 and trying to upgrade to 3.1.0. below is the command I use for starting the application 

java -jar -DCONDUCTOR_CONFIG_FILE=$config_file  conductor-server-*-SNAPSHOT-boot.jar  /app/config/log4j.properties

In the config properties file I have included 
conductor.additional.modules= files which are created extending the abstract module

Now that the newer version doesn't have conductor.additional.modules property as before in 2.31.4 https://github.com/Netflix/conductor/blob/v2.31.4/core/src/main/java/com/netflix/conductor/core/config/Configuration.java

So, how we can specify the additional modules with the newer version v3.1.0??

Thanks in advance.

**Comment by apanicker-nflx** [comment]
What are the additional modules that you are looking to add to the server? 
Please note that v3.x uses the SpringBoot framework while v2.31.4 used the Guice framework. 

As a result, you would need to convert your Guice modules to Spring configurations and then make these available in the component classpath. You would then need to explicitly enable scanning of these packages in the application [here](https://github.com/Netflix/conductor/blob/main/server/src/main/java/com/netflix/conductor/Conductor.java) using - 

```java
@SpringBootApplication(scanBasePackages = {"com.netflix.conductor", "com.additional.module.package"})
```

If you would like to simplify this approach and enable dynamic registration of additional spring beans, we would appreciate a new PR with the changes.
